### PR TITLE
changing the input arguments datatype (of summary stats) from data frame to dictionary

### PR DESCRIPTION
**Pull request by Ati-74**

### CODE REVIEW

File: Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py [hunk 1]
```diff
@@ -1,53 +1,86 @@
 """
```
Comment (cheekolegend) [inline Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py:1]: This density_calculation script seems to be duplicate code of the density calculation I'm writing elsewhere. The density calculation script has also been updated. Not sure how github deals with that, but maybe safer to unload this commit?

File: Scripts/Run-SS-On-Exp&Sim/scripts/growth_rate_stats.py [hunk 2]
```diff
@@ -1,39 +1,34 @@
-"""
+'''
```
Comment (cheekolegend) [inline Scripts/Run-SS-On-Exp&Sim/scripts/growth_rate_stats.py:1]: This growth_rate_stat script seems to be duplicate code of the calculation I'm writing elsewhere. The growth_rate script has also been updated. Not sure how github deals with that, but maybe safer to unload this commit?

File: Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py (hunk 1 above)
Comment (Ati-74) [reply Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py:1]: If I unload from this commit (assuming you mean removing it from the directory), it leads us to lose density calc. plus growth-rate-calc from the main script (due to not having the scripts in the directory to call). I think this is not what we want, right? so maybe the best is to talk in person to figure out what is efficient to do

File: Scripts/Run-SS-On-Exp&Sim/scripts/growth_rate_stats.py (hunk 2 above)
Comment (Ati-74) [reply Scripts/Run-SS-On-Exp&Sim/scripts/growth_rate_stats.py:1]: please see the above comment

File: Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py (hunk 1 above)
Comment (cheekolegend) [reply Scripts/Run-SS-On-Exp&Sim/scripts/density_calculation.py:1]: Sounds good. We can merge the directories we have been using in a future PR.

File: Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+def save_fig(cells, hull, points, fig_export_path, fig_name):
+    """
+    @param cells cellStates dict
+    @param hull object convex hull
+    @param points numpy array bacteria end points
+    @param  fig_export_path directory to export images
+    @param fig_name image name
+
+    """
+    # vertexes
```
Comment (cheekolegend) [inline Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py]: # vertices

File: Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    # find endpoints
+    endpoints_x, endpoints_y = cell_data.bacteria_end_points(cells)
+
+    points = np.column_stack((endpoints_x, endpoints_y))
+    hull = ConvexHull(points)
+
+    return hull, points
+
+
+def calc_fractal_dimension(cells, fig_export_path, fig_name):
```
Comment (cheekolegend) [inline Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py]: Should we calculate fractal dimension from the cell end-points or from from the colony border taken from an image? Worth discussing with @bpingalls.

**Review by cheekolegend — COMMENTED** [review]
Very nice images. Seems like the convex hull could capture the colony border better. Should discuss how to improve the fit if we proceed with getting the colony border using the convex hull algorithm.

**Review by cheekolegend — APPROVED** [review]
Looks good to me. Will let @bpingalls review and merge.

File: Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py (hunk 4 above)
Comment (bpingalls) [reply Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py]: Seems the image will be a more faithful representation

Comment (Ati-74) [reply Scripts/Run-SS-On-Exp&Sim/scripts/fractal_dimension.py]: @bpingalls Done!

File: Scripts/Run-SS-On-Exp&Sim/scripts/DataAnalysis.py [hunk 5]
```diff
@@ -117,17 +174,14 @@ def micro_colony_analysis(pickle_files_directory, summary_statistic_method, dt,
 
         # store ellipses
         ellipses = []
-        # store anisotropies of micro colonies of this time step
+        # store anisotropies of micro colonies of this rime step
```
Comment (cheekolegend) [inline Scripts/Run-SS-On-Exp&Sim/scripts/DataAnalysis.py]: time step

**Comment by cheekolegend** [comment]
Fractal dimension looks good to start testing. In future developments, try to put new features in separate PRs so that different features can be merged independently. Thanks!

**Pull request merged**